### PR TITLE
[@typespec/spector] - Enable Multiple Scenario Paths

### PR DESCRIPTION
**Pull request by sarangan12**
Currently we could execute the `serve`/`server-test`/... command such as:

`pnpm .\cmd\cli.mjs serve ..\http-specs\specs`

i.e we could provide only one path of scenarios to execute. But, since we separated the scenarios to azure and non-azure specific cases to 2 different packages, we need a way to execute commands such as:

`pnpm .\cmd\cli.mjs serve ..\http-specs\specs ..\..\..\typespec-azure\packages\azure-http-specs\specs\`

The same procedure must be followed in all other commands such as `validate-scenarios`, `generate-scenarios-summary`, etc. Since we have to do it for all the commands and we need to change in all the places, it is much simpler to do it in one location where we copy the files to one common temp location and execute from there. 

Also, in this PR, I have removed the `private: true` option for all the 5 packages to start the rollout process. 

Please review and approve the PR. Thanks

### CODE REVIEW

File: packages/spector/src/cli/cli.ts [hunk 1]
```diff
@@ -14,9 +15,56 @@ import { getCommit } from "../utils/misc-utils.js";
...
+  copyToTemp(scenariosPaths, `${tempDirname}`, `dist`);
+  copyToTemp(scenariosPaths, `${tempDirname}`, `assets`);
+  const tempDir = copyToTemp(scenariosPaths, `${tempDirname}`, `specs`);
+  return tempDir;
+}
+
 async function main() {
   await yargs(process.argv.slice(2))
-    .scriptName("spec")
+    .scriptName("tsp-spector")
```
Comment (sarangan12) [inline packages/spector/src/cli/cli.ts:67]: This is not really related to this change, But, I believe we could do this as part of this PR. Thanks

File: packages/http-specs/package.json [hunk 2]
```diff
@@ -1,6 +1,5 @@
 {
   "name": "@typespec/http-specs",
-  "private": true,
   "version": "0.1.0",
```
Comment (timotheeguerin) [inline packages/http-specs/package.json:3]: change the version to `0.1.0-alpha.0` too.

File: packages/spector/src/cli/cli.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
-import { join, resolve } from "path";
+import * as fs from "fs";
```
Comment (timotheeguerin) [inline packages/spector/src/cli/cli.ts:1]: import from `fs/promises` and try to no do `import * as fs`

File: packages/spector/src/cli/cli.ts [hunk 4]
```diff
@@ -35,29 +83,31 @@ async function main() {
       }
     })
     .command(
-      "validate-scenarios <scenariosPath>",
+      "validate-scenarios <scenariosPath..>",
```
Comment (timotheeguerin) [inline packages/spector/src/cli/cli.ts:86]: rename property to `scenariosPaths`

Comment (timotheeguerin) [reply packages/spector/src/cli/cli.ts:86]: all the way down teh line too

File: packages/spector/src/cli/cli.ts [hunk 5]
```diff
@@ -14,9 +15,56 @@ import { getCommit } from "../utils/misc-utils.js";
 
 export const DEFAULT_PORT = 3000;
 
+function copyDir(src: string, dest: string) {
```
Comment (timotheeguerin) [inline packages/spector/src/cli/cli.ts:18]: can we not just run it from where they are defined. This bring a lot of variability and makes it very hard to test any fix if the error show up in a temp directory

Comment (sarangan12) [reply packages/spector/src/cli/cli.ts]: I have removed the `copyDir` and optimized the code.

File: packages/spector/src/cli/cli.ts (hunk 4 above)
Comment (sarangan12) [reply packages/spector/src/cli/cli.ts]: Done

File: packages/spector/src/cli/cli.ts (hunk 3 above)
Comment (sarangan12) [reply packages/spector/src/cli/cli.ts]: Done

File: packages/spector/src/cli/cli.ts (hunk 5 above)
Comment (timotheeguerin) [reply packages/spector/src/cli/cli.ts]: Sorry I meant why do you copy things. We should run from the dir they are in

Comment (timotheeguerin) [reply packages/spector/src/cli/cli.ts]: Copying in a different dir is likely to break imports as well

Comment (sarangan12) [reply packages/spector/src/cli/cli.ts]: The reason why I copied the files is there are scripts such as `validate-scenarios`, etc will error out if they encounter any error in one of the scenario paths. Even the coverage was based on single commit. 

But, You are right. There is a possibility that the imports might break. With the recent update of PR, I have executed all scenarios as is in the current location.

Comment (sarangan12) [reply packages/spector/src/cli/cli.ts]: @timotheeguerin One important thing to note is the structure of `spec-coverage.json` has been modified to an array. So, this requires down the line changes. (Dashboard???)

Comment (timotheeguerin) [reply packages/spector/src/cli/cli.ts]: Can we not merge it as right now they are all unique. Otherwise I think the new structure should be a 2 level record with the first key being the test suite

Comment (sarangan12) [reply packages/spector/src/cli/cli.ts]: @timotheeguerin Even with the new structure they are still unique. It looks like this:

image

So, Do you want to do any changes in this?

Comment (timotheeguerin) [reply packages/spector/src/cli/cli.ts]: Eh I guess that's ok for now, we'll update when we update the dashboard

**Review by timotheeguerin — APPROVED** [review]
